File: service/jvm-java-quarkus-reactive/src/main/java/com/ecobenchmark/controllers/addtask/TaskResponse.java
package com.ecobenchmark.controllers.addtask;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.time.Instant;
import java.util.UUID;

public class TaskResponse {

    private UUID id;

    @JsonProperty("list_id")
    private UUID listId;

    private String name;

    private String description;

    @JsonProperty("creation_date")
    private Instant creationDate;

    public TaskResponse(UUID id, UUID listId, String name, String description, Instant creationDate) {
        this.id = id;
        this.listId = listId;
        this.name = name;
        this.description = description;
        this.creationDate = creationDate;
    }

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public UUID getListId() {
        return listId;
    }

    public void setListId(UUID listId) {
        this.listId = listId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Instant getCreationDate() {
        return creationDate;
    }

    public void setCreationDate(Instant creationDate) {
        this.creationDate = creationDate;
    }
}

File: service/php-symfony-apache2-dev/src/Controller/AddTask/TaskResponse.php
<?php

namespace App\Controller\AddTask;

use App\Entity\Task;

class TaskResponse
{
    public string $id;
    public string $list_id;
    public string $name;
    public string $description;
    public string $creation_date;

    static function fromTaskEntity(Task $task): TaskResponse
    {
        $taskResponse = new TaskResponse();
        $taskResponse->id = $task->getId();
        $taskResponse->list_id = $task->getList()->getId();
        $taskResponse->name = $task->getName();
        $taskResponse->description = $task->getDescription();
        $taskResponse->creation_date = $task->getCreationDate()->format('c');

        return $taskResponse;
    }
}
